docs/kb/semgrep-ci/github-reusable-workflows-semgrep.md

```diff
@@ -9,37 +9,32 @@ description: Setting up GitHub reusable workflows for Semgrep scans.
 
-1. Create a new repository with the reusable workflow:
+1. Create a new repository and add the `.github/workflows/semgrep.yml` file.
 
 ![image info](/img/kb/reusable-workflows-image-1.png)
 
-> Note: the semgrep.yml file must be in the `.github/workflows/` folder
+2. Add the desired job configuration to the `semgrep.yml` under `jobs:`. This can be the [recommended snippet](https://semgrep.dev/docs/semgrep-ci/sample-ci-configs/#sample-github-actions-configuration-file) or your current job configuration.
 
-2. The content of the file can be copied from the Semgrep [documentation](https://semgrep.dev/docs/semgrep-ci/sample-ci-configs/#sample-github-actions-configuration-file) but some changes need to be made in the `on:` statement. Basically, the condition to trigger the job must be in the caller, not here. So we can create the file as follows:
+3. Under the `on:` key, add `workflow_call`. This defines the condition to trigger the job described in the re-usable workflow: it's triggered by being called in other repositories.
 
 ![image info](/img/kb/reusable-workflows-image-2.png)
 
-3. Then, in each repository, you can call the reusable workflow as shown here:
+3. Then, in each repository, create or update the `semgrep.yml` to call the reusable workflow by modifying the `jobs:` key to use the reusable workflow. The easiest way to configure the `SEMGREP_APP_TOKEN` secret is to add it in the re-usable workflow and provide `secrets: inherit` in the calling workflow, as shown in the following example.
 
 ![image info](/img/kb/reusable-workflows-image-3.png)
 
-> Note: Secrets can be inherited from the reusable workflow and passed to the caller workflow.
-
 ## Running a scan
 
-Now, if a developer commits a change or makes a pull request (or otherwise in any way triggers the caller workflow), then the reusable Semgrep workflow will be called:
+Now, when a developer commits a change or makes a pull request (or otherwise in any way triggers the caller workflow), the reusable Semgrep workflow is called:
 
 ![image info](/img/kb/reusable-workflows-image-4.png)
 
-## Benefits
-
-1. As GitHub [says](https://docs.github.com/en/actions/using-workflows/reusing-workflows#overview): “Reusing workflows avoids duplication. This makes workflows easier to maintain and allows you to create new workflows more quickly by building on the work of others, just as you do with actions”.
-2. A reusable workflow can be triggered by several types of events, such as a push, pull_request, or schedule. In contrast, a required workflow can only be triggered by pull_request events.
-    
-## Drawbacks
+## Limitations
 
-1. Reusable workflows don’t avoid the creation of a `.github/workflows/semgrep.yml` file for each repo. You still need to create one to then insert a call to the reusable workflow.
+Reusable workflows don’t allow you to avoid the creation of a `.github/workflows/semgrep.yml` file for each repository. You still need to create one to call the reusable workflow.
```

Comment:
```suggestion
You still need to create a `.github/workflows/semgrep.yml` file for each repository to call the reusable workflow.
```
"don't allow you to avoid" sounds a bit awkward?

Reply:
I think this was probably intended to contrast it with required workflows, but it's probably more straightforward to say it positively given we don't know if the required workflows will continue to exist as such.